lib/logstash/inputs/tcp.rb

```diff
@@ -188,6 +188,12 @@ def decode_buffer(client_ip_address, client_address, client_port, codec, proxy_a
     end
   end
 
+  def flush_codec(codec, client_ip_address, client_address, client_port)
+    codec.respond_to?(:flush) && codec.flush do |event|
```

Comment:
this is not necessary, all codecs respond to `flush`, per https://github.com/elastic/logstash/blob/a3f6da3c3a61470ce510be0ea9ef2b6c07d47d5a/logstash-core/lib/logstash/codecs/base.rb#L73-L77  since a long time.